ios/FluentUI/Vnext/PersonaButtonCarousel/MSFPersonaButtonCarousel.swift

```diff
@@ -0,0 +1,110 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+import SwiftUI
+
+/// UIKit wrapper that exposes the SwiftUI PersonaGrid implementation
+@objc open class MSFPersonaButtonCarousel: NSObject, FluentUIWindowProvider, MSFPersonaCarouselState {
+
+    // MARK: - Public API
+
+    @objc open var view: UIView {
+        return hostingController.view
+    }
+
+    @objc public init(size: MSFPersonaButtonSize = .large,
+                      theme: FluentUIStyle? = nil) {
+        super.init()
+
+        personaButtonCarousel = PersonaButtonCarousel(size: size)
+
+        hostingController = UIHostingController(rootView: AnyView(personaButtonCarousel
+                                                                    .windowProvider(self)
+                                                                    .modifyIf(theme != nil, { personaButtonCarousel in
+                                                                        personaButtonCarousel.customTheme(theme!)
+                                                                    })))
+        hostingController.disableSafeAreaInsets()
+        view.backgroundColor = UIColor.clear
+    }
+
+    @objc public var count: Int {
+        return self.personaButtonCarousel.state.buttons.count
+    }
+
+    /// Adds a `PersonaButton` to the carousel, and returns an optional reference for additional property setting
+    ///
+    /// - Parameters:
+    ///   - primaryText: The primary text to appear in the `PersonaButton`
+    ///   - secondaryText: The secondary text to appear in the `PersonaButton`, below `primaryText`
+    ///   - image: The image to use as the persona's avatar
+    ///
+    /// - Returns: An optional reference to the added `PersonaButton`, which can be used to set additional properties or to update later
+    @discardableResult @objc public func add(primaryText: String?, secondaryText: String?, image: UIImage?) -> MSFPersonaButtonData {
+
+        let persona = MSFPersonaButtonStateImpl(size: self.state.buttonSize)
+
+        // Set passed-in properties
+        persona.primaryText = primaryText
```

Comment:
I think this logic should be in the state object and just called from here.
The scenario I'm thinking is a Carousel contained in another parent composite control. We would need to rely on the UIKit wrapper to add/remove PersonaButtons.

Reply:
Agreed. That said, I intentionally didn't address this until the team has had a chance to discuss our plan going forward for how to handle SwiftUI public interfaces. But for the time being, I think it makes sense to move all of these methods (add, remove, count, etc) to the `state` object and have this layer just pass through.